src/sdl3/gamepad.rs

```diff
@@ -121,23 +160,141 @@ impl GamepadSubsystem {
         }
     }
 
-    // FIXME:
-    // replaced with SDL_SetGamepadEventsEnabled() and SDL_GamepadEventsEnabled()
+    /// Return the implementation-dependent path of a gamepad.
+    /// This can be called before any gamepads are opened.
+    #[doc(alias = "SDL_GetGamepadPathForID")]
+    pub fn path_for_id(&self, joystick_id: JoystickId) -> Result<String, IntegerOrSdlError> {
+        use crate::common::IntegerOrSdlError::*;
+        let c_str = unsafe { sys::gamepad::SDL_GetGamepadPathForID(joystick_id) };
+
+        if c_str.is_null() {
```

Comment:
could use c_str_to_string_or_none

Reply:
`c_str_to_string_or_none` is the Option variant. This could be `c_str_to_string_or_err`, but it gives the Err as `Err(get_error())` and this one is returning the `Err(SdlError(get_error()))`.

I don't this this error should be `IntegerOrSdlError`, since the original API doesn't give integer as error, so I'll change the error to Error and use the `c_str_to_string_or_err`